src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -119,7 +126,8 @@ Keyword arguments
 function ShallowWaterModel(;
                            grid,
                            gravitational_acceleration,
-                      vertical_model = SingleLayerModel(),
+                vertical_formulation = SingleLayerModel(),
```

Comment:
can we imply this from the number of layers?

Reply:
Not entirely sure what you are suggesting - we do use `number_of_layers` in the function to imply the `vertical_formulation`. Do you suggest we can do it within the function declaration itself?

Reply:
@simone-silvestri is suggesting that the user should provide with the constructor `number_of_layers` and the constructor should put the right `vertical_model` formulation when constructing a model. Right?

Reply:
Okay, but the user implicitly mentions the number of vertical layers when creating a grid, so why ask them for the `number_of_layers` for the second time when we ourselves can get it from the grid creation?

Reply:
True. 

I think these are secondary details to deal later. Let’s first have a working model and we can finesse the user interface at the end.

Reply:
I agree, with all of that, there are `number_of_layers` and `vertical_formulation` in the constructor of the model. Both of them are redundant

Reply:
Thanks for the suggestion @simone-silvestri. It was just me being dense and not being able to follow - I've changed it and will push it.